process/smartContract/hooks/blockChainHook.go

```diff
@@ -404,6 +422,26 @@ func (bh *BlockChainHookImpl) NewAddress(creatorAddress []byte, creatorNonce uin
 func (bh *BlockChainHookImpl) ProcessBuiltInFunction(input *vmcommon.ContractCallInput) (*vmcommon.VMOutput, error) {
 	defer stopMeasure(startMeasure("ProcessBuiltInFunction"))
 
+	if bh.enableEpochsHandler.IsMaxBlockchainHookCountersFlagEnabled() {
+		gasSchedule := bh.gasSchedule.LatestGasSchedule()
```

Comment:
you should have your own gas schedule and not as a map - but save those 2 MAXs you need in separate parameters and change those when you have a gasSchedule change. Like we do in scProcessor.GasScheduleChange.

Reply:
done